lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -320,7 +321,31 @@ def _get(
             query = query.filter(Dataset.name.contains(json["name"]))
         if "filter" in json:
             query = self.filter_query(json["filter"], query)
-        query = self._build_sql_query(json.get("owner"), json.get("access"), query)
+
+        # The "mine" filter allows queries for datasets that are (true) or
+        # aren't (false) owned by the authenticated user. In the absense of
+        # the "mine" filter, datasets are selected based on visibility to the
+        # authenticated user and other explicit filters including "owner" and
+        # "access".
+        if "mine" in json:
+            if "owner" in json:
+                raise APIAbort(
+                    HTTPStatus.BAD_REQUEST,
+                    "'owner' and 'mine' filters cannot be used together",
+                )
+            auth_id = Auth.get_current_user_id()
+            if not auth_id:
+                raise APIAbort(
+                    HTTPStatus.BAD_REQUEST, "'mine' filter requires authentication"
+                )
+            if json["mine"]:
+                owner = auth_id
+            else:
+                owner = None
+                query = query.filter(Dataset.owner_id != auth_id)
+        else:
+            owner = json.get("owner")
+        query = self._build_sql_query(owner, json.get("access"), query)
```

Comment:
So if the `mine` is specified, `access` key is ignored, right? Do we also want to raise `BAD_REQUEST` in that case like you are doing for the owner/mine combination? Or should we also ignore the owner key if `mine` is specified?

Reply:
No; it works just like it did before. Obviously if you do `?mine=false&access=private` you won't find anything unless you're `ADMIN`, but that's what you asked for.

Similarly, `_build_sql_query` works with `?mine&access=private` (or public) to return "my" datasets with the specified access.

`mine=true` is exactly the same as `owner=<my_username>` in terms of the rest of the code; you just don't need to know the username string value.

`mine=false` is a bit funkier; but in terms of `_build_sql_query` it's not "no specification of user". If access also isn't specified, we end up with `access=public` because we're not allowed to see other private datasets. And we add the extra `query.filter(Dataset.owner_id != auth_id)`.